### PR TITLE
Pixel grid kwargs in DataAPI

### PR DESCRIPTION
**Pull request by smericks**
- Add option for user-defined pixel grid kwargs in DataAPI class with a new optional argument kwargs_pixel_grid
- Add corresponding data_api tests

### CODE REVIEW

**Review by sibirrer — CHANGES_REQUESTED** [review]
Looks good! I have some minor requests in terms of documentation and stability, but otherwise it's ready to merge this way

File: lenstronomy/SimulationAPI/data_api.py [hunk 1]
```diff
@@ -14,13 +14,17 @@ class DataAPI(SingleBand):
     options are available. Have a look in the specific modules if you are interested in.
 
     """
-    def __init__(self, numpix, **kwargs_single_band):
+    def __init__(self, numpix, kwargs_pixel_grid=None, **kwargs_single_band):
         """
 
         :param numpix: number of pixels per axis in the simulation to be modelled
+        :param kwargs_pixel_grid: if None, uses default pixel grid option
+            if defined, must contain keyword arguments for custom pixel grid
```
[the remark this answers is not in the document]
Comment (smericks) [reply lenstronomy/SimulationAPI/data_api.py:22]: Yes, done.

File: lenstronomy/SimulationAPI/data_api.py [hunk 2]
```diff
@@ -14,13 +14,17 @@ class DataAPI(SingleBand):
...
+            (ra_at_xy_0,dec_at_xy_0,transform_pix2angle)
         :param kwargs_single_band: keyword arguments used to create instance of SingleBand class
         """
         self.numpix = numpix
+        self.kwargs_pixel_grid = kwargs_pixel_grid
```
[the remark this answers is not in the document]
Comment (smericks) [reply lenstronomy/SimulationAPI/data_api.py:27]: Done.

File: lenstronomy/SimulationAPI/data_api.py [hunk 3]
```diff
@@ -39,13 +43,23 @@ def kwargs_data(self):
 
         :return: keyword arguments for ImageData class instance
         """
-        x_grid, y_grid, ra_at_xy_0, dec_at_xy_0, x_at_radec_0, y_at_radec_0, transform_pix2angle, transform_angle2pix = util.make_grid_with_coordtransform(
-            numPix=self.numpix, deltapix=self.pixel_scale, subgrid_res=1, left_lower=False, inverse=False)
+        # default pixel grid
+        if self.kwargs_pixel_grid is None:
+            _, _, ra_at_xy_0, dec_at_xy_0, _, _, transform_pix2angle, _ = (
```
[the remark this answers is not in the document]
Comment (smericks) [reply lenstronomy/SimulationAPI/data_api.py:48]: Reverted to previous formatting.

File: lenstronomy/SimulationAPI/data_api.py [hunk 4]
```diff
@@ -39,13 +43,23 @@ def kwargs_data(self):
...
+                util.make_grid_with_coordtransform(numPix=self.numpix, 
+                    deltapix=self.pixel_scale, subgrid_res=1, 
+                    left_lower=False, inverse=False) )
+        # user defined pixel grid
+        else:
+            ra_at_xy_0 = self.kwargs_pixel_grid['ra_at_xy_0']
```
[the remark this answers is not in the document]
Comment (smericks) [reply lenstronomy/SimulationAPI/data_api.py:54]: Yes I definitely should check this. Also added a test to check that it raises this error.

**Pull request merged**